src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -703,7 +703,10 @@ NetRouteMap FastRouteCore::getRoutes()
         int y2 = tile_size_ * (nodes[treeedge->n2].y + 0.5) + y_corner_;
         int l2 = nodes[treeedge->n2].botL;
         GSegment segment(x1, y1, l1 + 1, x2, y2, l2 + 1);
-        if (net_segs.find(segment) == net_segs.end()) {
+        // It is possible to have nodes that are not in adjacent layers so
```

Comment:
Elaborate on why it can have nodes that are not in adjacent layers.

Reply:
If an edge with length 0 is between a pin and a steiner node, since this edge is never analyze the attributes botL will have the default value that is the max routing layer. In these cases the steiner node considers the pin layer on it's stack alias and the other edges that use the stiener node will consider the correct pin layer.

Reply:
Could you add this explanation (perhaps more simplified) to the code? I want to make sure we'll have this well-documented in the code for future reference.